roxie/ccd/ccdprotocol.cpp

```diff
@@ -283,13 +283,13 @@ class ProtocolSocketListener : public ProtocolListener
             Owned<ISecureSocket> ssock;
             if (client)
             {
-                if (protocol && streq(protocol, "ssl"))
+                if (protocol.get() && streq(protocol.get(), "ssl"))
                 {
 #ifdef _USE_OPENSSL
                     try
                     {
                         if (!secureContext)
-                            secureContext.setown(createSecureSocketContextEx(certFile, keyFile, passPhrase, ServerSocket));
+                            secureContext.setown(createSecureSocketContextEx(certFile.get(), keyFile.get(), passPhrase.get(), ServerSocket));
```

Comment:
if createSecureSocketContextEx can handle nullptrs then this is fine.  If createSecureSocketContextEx can only handle "" then use .str()